WordPress/src/main/java/org/wordpress/android/WordPress.java

```diff
@@ -286,6 +289,14 @@ public void onLog(T tag, LogLevel logLevel, String message) {
         mCredentialsClient.connect();
     }
 
+    private void sanitizeMediaUploadStateForSite() {
+        int siteLocalId = AppPrefs.getSelectedSite();
+        SiteModel selectedSite = mSiteStore.getSiteByLocalId(siteLocalId);
+        if (selectedSite != null) {
+            UploadService.sanitizeMediaUploadStateForSite(mMediaStore, mDispatcher, selectedSite);
```

Comment:
I think performance is important since this is being called at startup and resume. Can this routine run in the background?

Also, we can exit [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java#L379) if both `uploadingMedia` and `queuedMedia` are empty (which will most likely be the case).

Reply:
background running addressed  in 933d71b
addressed the empty arrays case in 9f69224